pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -740,6 +740,7 @@ func TestSchedulingQueue_Close(t *testing.T) {
 		},
 	}
 	for _, test := range tests {
+		test := test
```

Comment:
`wg.Wait()` at line 758 will wait until the goroutine finishes, so there will be no problem?

Reply:
@hex108 thanks for the review. So, the issue is that:
When a go loop launches goroutines, it is a common mistake to assume
that the routines can reference the loop value from their specific loop
iteration. Actually, all goroutines share the same reference to the loop
veriable, and they typically all see the last value. To correct this, the loop variable must be captured as a parameter to the anonymous func. So the copy of test is needed to pass on to the goroutine in order for it to work properly. An example below. Run the following code and then run uncomment the lines: //k := k and //v := v and run again.

package main

import (
	"fmt"
	"time"
)
func main() {
	fmt.Println("Hello, playground")
	a := []int{1,2,3,4,5}
	for k, v := range a {
		//k := k 
		//v := v
		go func() {
			fmt.Println(k, v)
		} ()
	}
	
	time.Sleep(time.Second * 5)
}

Reply:
Yes, you are right. Howerver for this short test case, all goroutines run one by one.

Reply:
Yes, you are right: t.Run() will make it sequential (synchronous) rather than out of order (asynchronous). Thanks for pointing out.